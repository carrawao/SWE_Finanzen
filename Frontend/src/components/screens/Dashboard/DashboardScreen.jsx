import React, {useState} from 'react';
import ScreensTemplate from '../../ScreensTemplate';
import {Grid, Box} from '@mui/material';
import PropTypes from 'prop-types';
import AllocationGraph from './AllocationGraph';
import PortfolioOverview from './PortfolioOverview';


const DashboardScreen = (props) => {

  const renderBody = () => (
    
    <React.Fragment>
      <Grid 
        container
        direction="column"
        justifyContent="center"
        alignItems="space-evenly"
        marginTop= "10px"
      >
        <Grid
          container
          direction="row"
          justifyContent="space-evenly"
          alignItems="center"
        >
          <Grid item xs={4}>
            <AllocationGraph
              portfolioData={props.portfolioData[props.activePortfolio]}
              getAllAssets={props.getAllAssets}
            />
          </Grid>
          <Grid item xs={8}>
            <PortfolioOverview
              portfolioData={props.portfolioData}
              activePortfolio={props.activePortfolio}
              setActivePortfolio={props.setActivePortfolio}
            />
          </Grid>
        </Grid>
        <Grid
          container
          direction="row"
          justifyContent="space-evenly"
          alignContent="center"
        >
          <Grid item xs={6}>
            
          </Grid>
          <Grid item xs={6}>
            
          </Grid>
        </Grid>
      </Grid>
    </React.Fragment>
    );

  return (
    <React.Fragment>
      <ScreensTemplate
        bodyComponent={renderBody}
        selectedNavLinkIndex={1}
        assetsListArray={props.assetsListArray}
        searchResult={props.searchResult}
        setSearchResult={props.setSearchResult}
      />
    </React.Fragment>

  );
}

DashboardScreen.propTypes = {
  activePortfolio: PropTypes.string,
  setActivePortfolio: PropTypes.func,
  portfolioData: PropTypes.object,
  setPortfolioData: PropTypes.func,
  getAllAssets: PropTypes.func,
  searchResult: PropTypes.array,
  setSearchResult: PropTypes.func,
  watchListsArray: PropTypes.array,
  assetsListArray: PropTypes.array
};

export default DashboardScreen;