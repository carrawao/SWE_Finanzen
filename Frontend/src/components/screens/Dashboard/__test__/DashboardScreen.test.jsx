import React from 'react';
import DashboardScreen from '../DashboardScreen';
import toJson from 'enzyme-to-json';
import Enzyme, {shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({adapter: new Adapter()})


it('DashboardScreen renders without crashing', () => {
  const portfolioData = {
    "Portfolio": {
      "name": "Portfolio",
      "value": 0,
      "gains": 0,
      "realisedGains": 0,
      "performance": 0,
      "shares": [],
      "crypto": [],
      "cash": [],
      "activities": [],
      "updated": "1970-01-01"
    },
  };

  const setSearchResult = jest.fn();
  const setPortfolioData = jest.fn();
  const setActivePortfolio = jest.fn();
  const getAllAssets = jest.fn();
  const setStatusMessage = jest.fn();
  const setMessageType = jest.fn();

  const wrapper = shallow(
    <DashboardScreen
      searchResult={[]}
      setSearchResult={setSearchResult}
      watchListsArray={[]}
      assetsListArray={[[]]}
      activePortfolio={'Portfolio'}
      setActivePortfolio={setActivePortfolio}
      getAllAssets={getAllAssets}
      portfolioData={portfolioData}
      setPortfolioData={setPortfolioData}
      setStatusMessage={setStatusMessage}
      setMessageType={setMessageType}
    />);

  expect(toJson(wrapper)).toMatchSnapshot();
})