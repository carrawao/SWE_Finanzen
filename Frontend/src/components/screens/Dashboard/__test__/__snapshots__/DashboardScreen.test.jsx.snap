// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardScreen renders without crashing 1`] = `
<Fragment>
  <ScreensTemplate
    assetsListArray={
      Array [
        Array [],
      ]
    }
    bodyComponent={[Function]}
    searchResult={Array []}
    selectedNavLinkIndex={1}
    setMessageType={[MockFunction]}
    setSearchResult={[MockFunction]}
    setStatusMessage={[MockFunction]}
  />
</Fragment>
`;
